ipaserver/plugins/baseuser.py

```diff
@@ -602,6 +607,11 @@ def check_objectclass(self, ldap, dn, entry_attrs):
                     answer = self.api.Object['radiusproxy'].get_dn_if_exists(cl)
                     entry_attrs['ipatokenradiusconfiglink'] = answer
 
+            if 'ipatokenradiususername' in entry_attrs:
+                if 'ipatokenradiusproxyuser' not in obj_classes:
+                    entry_attrs['objectclass'].append(
+                        'ipatokenradiusproxyuser')
+
```

Comment:
Isn't this functionally equivalent to lines 488-490, calling add_missing_object_class()?

Reply:
Yes, it is. IMHO both places should look like this:

```
if 'ipatokenradiususername' in entry_attrs:
    add_missing_object_class(ldap, u'ipatokenradiusproxyuser', dn,
                             entry_attrs, update=False)
```

Reply:
Lines 488-490 are called for user-add, while these lines are called for user-mod.
In user-add (new code), I chose to call the existing function add_missing_object_class, but here I kept the same style as the surrounding lines. The rationale was that obj_classes was already fetched and lowercase'd in line 590, so it looked more efficient. But I can use add_missing_object_class here, too, if you think it's better.